CMakeLists.txt

```diff
@@ -329,19 +318,22 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR}/Userland/Services)
 include_directories(${CMAKE_CURRENT_BINARY_DIR}/Userland/Libraries)
 include_directories(${CMAKE_CURRENT_BINARY_DIR}/Userland)
 
+include(Meta/CMake/wasm_spec_tests.cmake)
+set(write_if_different ${SERENITY_PROJECT_ROOT}/Meta/write-only-on-difference.sh)
```

Comment:
Uhh hmm. That does look to do the same thing that our little helper script does. @BenWiederhake  @nico  is there a technical reason we didn't use `cmake -E copy_if_different` when adding that script and the first few uses of it?

Reply:
Oh I think our uses of that script would need to change a bit. the script can parse a bunch of arguments to create the file, and executes $2 with $3...N. 
So we have like
```
    add_custom_command(
        OUTPUT ${output}
        COMMAND ${write_if_different} ${output} $<TARGET_FILE:Lagom::IPCCompiler> ${source}
        VERBATIM
        DEPENDS Lagom::IPCCompiler
        MAIN_DEPENDENCY ${source}
    )
```
which would need to change to ... something else? To use ${CMAKE_COMMAND} -E copy_if_different, right? Unless there's an obvious extra argument to add_custom_command() that we could use.

Reply:
Your example could be rewritten as:
```cmake
add_custom_command(
    OUTPUT "${output}"
    BYPRODUCTS "${output}.tmp"
    COMMAND "$<TARGET_FILE:Lagom::IPCCompiler>" "${source}" > "${output}.tmp"
    COMMAND "${CMAKE_COMMAND}" -E copy_if_different "${output}.tmp" "${output}"
    VERBATIM
    MAIN_DEPENDENCY "${source}"
)
```

Reply:
The reason for `write-only-on-difference.sh` is that I didn't know that `copy_if_different` exists :D

If this were for many thousands of files, I'd worry whether this approach is measurably slower, because `write-only-on-difference.sh` might cache more efficiently. However, that's probably a non-issue.

Reply:
Well... One difference `write-only-on-difference.sh` *does* have is that it cleans up all those byproduct files after it creates them. And if we look at, for example, all the JavaScript wrapper bindings we generate...
```
:~/serenity/Build/i686$ find ./Userland/Libraries/LibWeb -name "*.tmp" | wc -l
672
```
There sure are a lot of them 😅 .

```
~/serenity/Build/i686$ find . -name "*.tmp" | wc -l
753
```

Looking at 753 `.tmp` headers/cpp files laying around in your build directory once we add in the gml files and the IPC files. Which I have no idea if the filesystem usage there at rest matters (probably not? idk) or if the fact that we are overwriting tmp files rather than regenerating them entirely for partial rebuilds matters more.

Reply:
```
 find . -name "*.tmp" -print0 | du --files0-from=- -ch | tail -1
4.8M    total
```

5 Megabytes? meh.

Though there are some noticeable chonkers in there:
```
40K     ./Userland/Services/LaunchServer/LaunchServerEndpoint.h.tmp
52K     ./Userland/Services/AudioServer/AudioServerEndpoint.h.tmp
64K     ./Userland/Services/WebContent/WebContentServerEndpoint.h.tmp
68K     ./Userland/Libraries/LibWeb/Bindings/HTMLElementPrototype.cpp.tmp
88K     ./Userland/Libraries/LibWeb/Bindings/DocumentPrototype.cpp.tmp
88K     ./Userland/Services/WebContent/WebContentClientEndpoint.h.tmp
100K    ./Userland/Services/WindowServer/WindowClientEndpoint.h.tmp
184K    ./Userland/Libraries/LibVT/EscapeSequenceStateMachine.h.tmp
284K    ./Userland/Services/WindowServer/WindowServerEndpoint.h.tmp
```

Reply:
It probably doesn't make a big difference. But if you really care, maybe it can be cleaned up by an additional
`COMMAND rm "${output}.tmp"` or something?

Reply:
Yeah, it's not a requirement to keep byproducts around. Removal should preferably happen with `"${CMAKE_COMMAND}" -E remove "${output}.tmp"`.